repo/iRepository.ts

```diff
@@ -0,0 +1,89 @@
+import { v4 as uuid } from 'uuid';
+
+//define Task data values and types
+interface Task {
+    id: uuid;
+    name: string;
+    createdAt: string;
+    dueDate: string;
+    completed: boolean;
+}
+
+//generic repository interface
+interface IRepo<Task>{
+    add(T: Task)
+    getId(id: uuid)
+    sortBy(val: string)
+    updateById(id: string, updated: Task)
+    removeById(id: string)
+}
+
+
+//implementation of generic repository interface 
+class taskRepository implements IRepo<Task>  {
+
+    //Not using a db yet so create local array to store vals
+    task_list: Task[] = []
+    
+    public add(T){
+        T.payload.id = uuid()
+        this.task_list.push(T.payload)
+        return T.payload
+    }
+
+    public getId(new_id){
+        if(this.task_list.find(({id}) => id === new_id)){
+            return this.task_list.find(({id}) => id === new_id)
+        }else{
+            return "No task with ID specified found "
```

Comment:
Why are you returning a string rather than nil or something? Try to strongly type the return of this function I think you'll see what the issue is.